app/controllers/users.rb

```diff
@@ -0,0 +1,18 @@
+get '/users/new' do
+  @user = User.new
+  erb :'users/new'
+end
+
+post '/users' do
+  name, username, email, password, password_confirmation = params[:name], params[:username],params[:email], params[:password], params[:password_confirmation]
+  @user = User.create(name: name, username: username, email: email, password: password, password_confirmation: password_confirmation)
```

Comment:
Consider passing the params hash itself to User.create, rather than extracting out individual variables. Of course, you then have the problem of there possibly being form data in the params hash that you do not want (after all users can always submit form data that you do not intend them to). The rails solution to this is the following: http://apidock.com/rails/ActionController/Parameters/permit
You could do something similar in sinatra, i.e have a helper method within each controller that filters the params for only the values needed to create the relevant resource.